api/v1alpha2/inferencepool_types.go

```diff
@@ -157,6 +159,10 @@ type InferencePoolStatus struct {
 	//
 	// +kubebuilder:validation:MaxItems=32
 	Parents []PoolStatus `json:"parent,omitempty"`
+	//ReadyPods is the number of pods that are associated with the InferencePool and are ready
+	// to serve inference requests. A pod is considered ready if its Ready condition is set to
+	// "True" and it is not marked for deletion.
+	ReadyPods int `json:"readyPods,omitempty"`
```

Comment:
What is the goal for this field? Historically in Kubernetes we've tried to ensure a separation of concerns here. Ideally the portion of a Gateway implementation that handles endpoint readiness should be something that can be disconnected from higher level routing config. 

More broadly, I'd be worried that trying to keep track of this in status would dramatically increase the number of updates to InferencePool status and the overall load on the controller. Instead of needing to reconcile when an InferencePool changed, it would need to reconcile anytime the _readiness_ or _number_ of connected endpoints changed.

Reply:
+1 to ^. I prefer continuing to use the `ResolvedRefs=False` condition to indicate the following:

1. `spec.selector` returns `0` Pods.
2. `ExtensionReference` can be resolved.

@robscott @kfswain @ahg-g is my understanding of `ResolvedRefs` consistent with yours?